Add support for SSL Cipher lists

dalees opened the pull request:
This patch adds optional array parameter 'ssl_ciphers' which go into rabbitmq's config if SSL is enabled and limit the ciphers to the specified list.

Documentation is erlang as these parameters are passed through: http://www.erlang.org/doc/man/ssl.html
Supported ciphers can be listed with: rabbitmqctl eval 'ssl:cipher_suites().'
Functionality can be tested with something like Cipherscan: https://github.com/jvehent/cipherscan.git

dalees commented:
Spotted a syntax error. Updated patch coming shortly.

jonnytpuppet commented:
Thank you for your contribution - it looks good!

It would be appreciated if you could update the README with information on the new property before this can be merged.

Also, please remember to squash your commits.

Thanks!

dalees commented:
Cheers. Rebased to master, added a section on ssl_ciphers to README.md and squashed.

Includes the above helpful tips to list ciphers and test functionality. I knew there was a better place to put that than in the pull request!

jonnytpuppet commented:
Hi There

Unfortunately the tests have failed. It would be appreciated if you could take a look at this and re-submit code that fixes things before this can be merged.

Thanks!

dalees commented:
Huh, strange. Looks like an output formatting change in the spec tests, the same issue appears on master travis: https://travis-ci.org/puppetlabs/puppetlabs-rabbitmq/builds

I think I've got a solution; created #335 to try and resolve.
